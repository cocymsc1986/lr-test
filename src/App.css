.app {
  display: flex;
}

.sidebar {
  width: 20%;
  min-width: 150px;
  border-right: 3px solid purple;
  height: 100vh;
  font-size: 20px;
  padding: 12px;
}

.results {
  width: 70%;
  padding: 20px;
}

.header {
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 12px;
}

.filters {
  margin-top: 12px;
}

.filters__option {
  margin-right: 6px;
  transform: translateY(-20%);
}

.result {
  margin-bottom: 30px;
  padding: 12px 12px 24px;
  border-bottom: 1px solid lightgrey;
}

.result__title {
  margin-bottom: 6px;  
}

.result__stars {
  margin: 6px 6px 6px 0;
}

.result__facility {
  display: inline-block;
  padding: 3px 6px;
  color: white;
  background: purple;
  margin-right: 4px;
  border-radius: 3px;
}
