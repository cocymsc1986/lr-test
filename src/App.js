import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          Search Results
        </header>
        {/* MAIN */}
      </div>
    );
  }
}

export default App;
