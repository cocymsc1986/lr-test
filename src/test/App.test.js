import React from 'react';
import { App } from '../App';

import resultsData from '../helpers/search-data'; 

it('renders', () => {
  const wrapper = shallow(<App />);

  expect(wrapper).toMatchSnapshot();
});

describe('updateFilters should update filteredResults state', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({ results: resultsData });
  wrapper.instance().updateFilters({ target: { checked: true, value: 'pool' } });
  
  expect(wrapper.state().filters).toEqual(['pool']);
  expect(wrapper.state().filteredResults).toEqual([
    {
      "name": "hotelone",
      "starRating": 5,
      "facilities": ["car park", "pool"]
    }
  ]);
});

describe('updateSort with ascend should sort filteredResults ascending', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({ filteredResults: resultsData });
  wrapper.instance().updateSort({ target: { value: 'ascending' }});

  expect(wrapper.state().filteredResults).toEqual([
    {
      "name": "hoteltwo",
      "starRating": 3,
      "facilities": ["car park", "gym"]
    },
    {
      "name": "hotelthree",
      "starRating": 3,
      "facilities": []
    },
    {
      "name": "hotelone",
      "starRating": 5,
      "facilities": ["car park", "pool"]
    }
  ])
});

describe('sortResults with descend should sort filteredResults descending', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({ filteredResults: resultsData });
  wrapper.instance().updateSort({ target: { value: 'descending' }});

  expect(wrapper.state().filteredResults).toEqual([
    {
      "name": "hotelone",
      "starRating": 5,
      "facilities": ["car park", "pool"]
    },
    {
      "name": "hoteltwo",
      "starRating": 3,
      "facilities": ["car park", "gym"]
    },
    {
      "name": "hotelthree",
      "starRating": 3,
      "facilities": []
    }
  ])
});