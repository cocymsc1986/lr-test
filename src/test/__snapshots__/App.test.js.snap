// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className=""
        >
          Search Results
        </header>,
        <div>
          <_default
            updateFilters={[Function]}
          />
        </div>,
        Array [],
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search Results",
          "className": "",
        },
        "ref": null,
        "rendered": "Search Results",
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <_default
            updateFilters={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "updateFilters": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className=""
          >
            Search Results
          </header>,
          <div>
            <_default
              updateFilters={[Function]}
            />
          </div>,
          Array [],
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Results",
            "className": "",
          },
          "ref": null,
          "rendered": "Search Results",
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <_default
              updateFilters={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "updateFilters": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
